### ISSUE TITLE
Reconnect Button broken with patch 27.4

### ISSUE THREAD

**Issue by plivanos** [haoruan/HDT-Reconnector#23]
Hello, patch 27.4 broke the reconnect button again!

**Comment by hizenzen** [comment]
true dat

**Comment by NotThatMiko** [comment]
+1

**Comment by Splendent** [comment]
nah...HDT remove it from available plugin, no idea why

**Comment by tartru-std** [comment]
In the commit they said "Blizzard has kindly asked us to stop supporting this plugin" 
They are juste doing a check on the plugin name. I suppose that you can just change the plugin name and it will work again.
HDT commit : https://github.com/HearthSim/Hearthstone-Deck-Tracker/commit/157187be463827ce4d64e5630c7096faebc4d0e2

**Comment by hizenzen** [comment]
Fuck blizzard and their slow ass animations. They want to force us watching
their shit,but most people only care about the end result and don't want to
waste time watching the animations

**Comment by tartru-std** [comment]
Use a random value instead of an hardcoded string for the plugin name and it will do the trick

**Comment by Koen-Moore** [comment]
> Use a random value instead of an hardcoded string for the plugin name and it will do the trick

What is the easiest way to do that? I tried changing the name of the DLL and it didn't work

**Comment by maxrenke** [comment]
the dll needs to be recompiled with a new name... if only haoruan shared build instructions 🤔

**Comment by SaveZelda** [comment]
> Fuck blizzard and their slow ass animations. They want to force us watching their shit,but most people only care about the end result and don't want to waste time watching the animations
> […](#)

Blizzard is a dumbass company, period. The only thing they know is how to ruin players experience and destroy their own IPs

**Comment by st0rch** [comment]
I got it to work in like 10 minutes. 



1. Clone the repo

2. In the ReconnectorPlugin.cs file, change the value of the "Name" property to whatever you want
    -   I changed any class name referencing "Reconnector" to be safe, but It's almost certainly just this one property as the above commit from the HSDeckTracker devs showed the code looking for plugin.name, and this file contains all of the info about the plugin and inherits from their plugin interface.
    
3. Build the project, which will put the DLL file (and a bunch of other garbage) in your Bin folder.

4. Drop it in the HDT plugin folder and you're good to go. 

I can share the rebuilt plugin, or if you'd rather rebuild it just follow these instructions.

**Comment by hizenzen** [comment]
Yeah, I got it working as well.

**Comment by Doud78** [comment]
Hey storch, how do you build the project ?

**Comment by hizenzen** [comment]
Visual Studio

**Comment by Doud78** [comment]
yes, but how, like i ve put the project in VS and then what u do? sry for those stupid questions

**Comment by hizenzen** [comment]
Control+B or or go to Build/Build HDT-Reconnector

**Comment by ladragee** [comment]
works fine thx

**Comment by Reachground** [comment]
Can you share it @st0rch?

**Comment by Shaplow** [comment]
Yeah it will be nice if u can share it @st0rch, i try to rebuild it but i fail aha

**Comment by hizenzen** [comment]
Here, this is a clone of haoruan's repo( I checked the license, this should be fine, but if haoruan has any problems with it, I can delete it right away).

https://github.com/hizenzen/HSReconnect/tree/HS_Reconnector

P.S I have also temporarily disabled the "Simulate" option, so there is only a "Reconnect" button now.

**Comment by meliodas991023** [comment]
hizenzen the link you shared does not work

**Comment by hizenzen** [comment]
My bad, it was private, changed it to public

**Comment by meliodas991023** [comment]
thanks

**Comment by hizenzen** [comment]
btw, it seems like disabling the simulate broke something so there is a "key" text instead of reconnect, I'll fix this later

**Comment by TZUMITNEM** [comment]
Thanks a lot @hizenzen

**Comment by Reachground** [comment]
Awesome @hizenzen. Thanks! The "Key:" is funny but it reconnects!

EDIT: Pressed it and after that it's just grey. I really don't care that much, I'm just happy it works again!

**Comment by Shaplow** [comment]
Thanks a lot !! 🔥Le 21 sept. 2023 à 12:50, Reachground ***@***.***> a écrit :﻿
Awesome @hizenzen. Thanks! The "Key:" is funny but it reconnects!

**Comment by hizenzen** [comment]
The reconnect button should be fixed now(download v1.4.8).
Anyway, renaming the plugin is  just a temporary fix, I think HS deck tracker will find a better way to block this plugin soon .

**Comment by SaveZelda** [comment]
> The reconnect button should be fixed now(download v1.4.8). Anyway, renaming the plugin name is just a temporary fix, I think HS deck tracker will find a better way to block this plugin soon .

Thanks so much hizenzen! The hero we need indeed :)

**Comment by kuroanon** [comment]
the zip from hizenzen is flagged by my AV

**Comment by hizenzen** [comment]
> the zip from hizenzen is flagged by my AV

Then I guess haoruan's plugin is a virus as well.

Seriously, if you think it's a virus, simply don't use it. I only created this repo to help the other folks who can't compile the source code with VS.

**Comment by Reachground** [comment]
Installed the latest version. Thanks again @hizenzen! 🏆

Scanned it with my AV from work. Nothing as expected, kuroanon has to stop using Blizzard's Malware protection 😆

**Comment by daakal** [comment]
you can just open the code in github and view it, its very short, its not a virus :) thank you hizenzen for the quick fix so i can get back to not rushing between turns 👍

**Comment by haoruan** [comment]
> Here, this is a clone of haoruan's repo( I checked the license, this should be fine, but if haoruan has any problems with it, I can delete it right away).
> 
> https://github.com/hizenzen/HSReconnect/tree/HS_Reconnector
> 
> P.S I have also temporarily disabled the "Simulate" option, so there is only a "Reconnect" button now.

No problem at all, feel free to update it and thanks for making it work again.

I was notified several weeks ago: "HDT Reconnector plugin is not something they would like to exist for Hearthstone", and was asked to take it down and stop offering it. So I plan to remove the DLL files in the coming weeks but will still keep the source code so that everyone can compile the DLL themselves.

At last, this is not a functional bug and HDT may block this plugin again, so I won't fix it in this repo, you can refer to above comments for a workaround.

**Comment by Mentalist069** [comment]
> Here, this is a clone of haoruan's repo( I checked the license, this should be fine, but if haoruan has any problems with it, I can delete it right away).
> 
> https://github.com/hizenzen/HSReconnect/tree/HS_Reconnector
> 
> P.S I have also temporarily disabled the "Simulate" option, so there is only a "Reconnect" button now.

stopped working. please update )

**Comment by Reachground** [comment]
Oh, I misunderstood. Downloaded that version but it didn't work. Of course not, it's the old version. Well let's see if someone can get it to work again. It's so annoying to see the slow fights.

**Comment by Mentalist069** [comment]
oh, yes). I really hope that @hizenzen will make an updated version

**Comment by geberten** [comment]
if u dont wanna wait  for @hizenzen uninstall hdt  and  go to [https://github.com/HearthSim/Hearthstone-Deck-Tracker/releases](url)  and dowland [v1.22.10] and make sure u close check for uptades.

**Comment by Mentalist069** [comment]
> если ты не хочешь ждать@hizenzenудалите hdt и перейдите по адресу [https://github.com/HearthSim/Hearthstone-Deck-Tracker/releases](url) и dowland [v1.22.10] и убедитесь, что вы закрыли проверку на наличие обновлений.

Thanks I'll try

**Comment by Mentalist069** [comment]
> > если ты не хочешь ждать@hizenzenудалите hdt и перейдите по адресу [https://github.com/HearthSim/Hearthstone-Deck-Tracker/releases](url) и dowland [v1.22.10] и убедитесь, что вы закрыли проверку на наличие обновлений.
> 
> Thanks I'll try

yes, it works that way, thank you!
